src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java

```diff
@@ -197,23 +202,35 @@ public AnnotationVisitor visitTypeAnnotation(int typeRef, TypePath typePath, Str
 		}
 
 		@Override
-		public AnnotationVisitor visitAnnotationDefault() {
-			return AsmClassRemapper.createAsmAnnotationRemapper(Type.getObjectType(owner).getDescriptor(), super.visitAnnotationDefault(), remapper);
+		public AnnotationVisitor visitAnnotationDefault() { // TODO: should we pass Type.getObjectType(owner).getDescriptor() or null for descriptor?
+			try {
+				AnnotationVisitor annotationVisitor = (AnnotationVisitor) MethodVisitor.class.getMethod("visitAnnotationDefault").invoke(this);
+				return annotationVisitor == null
+						? null
+						: createAnnotationRemapper(Type.getObjectType(owner).getDescriptor(), annotationVisitor);
+			} catch (Exception e) {
+				throw new RuntimeException("This should not happen!", e);
+			}
```

Comment:
Why do we pass `Type.getObjectType(owner).getDescriptor()` instead of `null`. This confuses me considering the comment in 792 (https://github.com/LogicFan/tiny-remapper/blob/0dac35458491b163a1ba0d382a7675a93d3b3b70/src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java#L792)